### PR TITLE
teddycloud

### PR DESCRIPTION
**Pull request by catduckgnaf**
https://github.com/toniebox-reverse-engineering/teddycloud

### CODE REVIEW

File: templates/teddycloud.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <Support>https://t.me/toniebox_reverse_engineering</Support>
+    <Project>https://github.com/toniebox-reverse-engineering/teddycloud</Project>
+    <Overview>teddyCloud is an alternative server for your Toniebox, allowing you to host the cloud services locally. This gives you the control about which data is sent to the original manufacturers cloud and allows you to host your own figurine audio files on e.g. your NAS or any other server.
+    </Overview>
+    <WebUI>http://[IP]:[PORT:7780]</WebUI>
+    <Category>MediaServer:Music MediaServer:Other</Category>
+    <TemplateURL>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/teddycloud.xml</TemplateURL>
+    <Icon>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/img/teddycloud.xml</Icon>
+    <Config Name="port 80 (optional)" Target="80" Default="" Mode="tcp" Description="optional" Type="Port" Display="always" Required="false" Mask="false">7780</Config>
+    <Config Name="443" Target="443" Default="" Mode="tcp" Description="" Type="Port" Display="always-hide" Required="true" Mask="false">7443</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:17]: I think both these ports should be always visible and required. Also, please set their `Default`s and use more descriptive `Name`s.

File: templates/teddycloud.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <Config Name="content" Target="/content" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/content</Config>
+    <Config Name="config" Target="/config" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/teddycloud/config</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:19]:
```suggestion
    <Config Name="config" Target="/teddycloud/config" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/teddycloud/config</Config>
```

File: templates/teddycloud.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <Config Name="certs" Target="/certs" Default="" Mode="rw" Description="Place your certs here after grabbing them off your device. (Yes you need to do this)" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/certs</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:20]:
```suggestion
    <Config Name="certs" Target="/teddycloud/certs" Default="" Mode="rw" Description="Place your certs here after grabbing them off your device. (Yes you need to do this)" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/certs</Config>
```

File: templates/teddycloud.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <Config Name="library" Target="/library" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/library</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:21]:
```suggestion
    <Config Name="library" Target="/teddycloud/data/library" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/library</Config>
```

File: templates/teddycloud.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <Overview>teddyCloud is an alternative server for your Toniebox, allowing you to host the cloud services locally. This gives you the control about which data is sent to the original manufacturers cloud and allows you to host your own figurine audio files on e.g. your NAS or any other server.
+    </Overview>
+    <WebUI>http://[IP]:[PORT:7780]</WebUI>
+    <Category>MediaServer:Music MediaServer:Other</Category>
+    <TemplateURL>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/teddycloud.xml</TemplateURL>
+    <Icon>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/img/teddycloud.xml</Icon>
+    <Config Name="port 80 (optional)" Target="80" Default="" Mode="tcp" Description="optional" Type="Port" Display="always" Required="false" Mask="false">7780</Config>
+    <Config Name="443" Target="443" Default="" Mode="tcp" Description="" Type="Port" Display="always-hide" Required="true" Mask="false">7443</Config>
+    <Config Name="content" Target="/content" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/content</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:18]:
```suggestion
    <Config Name="content" Target="/teddycloud/data/content" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/content</Config>
```

Looking at [the Dockerfile for the container](https://github.com/toniebox-reverse-engineering/teddycloud/blob/e04ed1a36371c52cd229bec91d2364633f3ec30d/DockerfileUbuntu#L39), it looks like this is expected to map to `/teddycloud/data/content`, please update the `Target` accordingly.

File: templates/teddycloud.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <Config Name="config" Target="/config" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/teddycloud/config</Config>
+    <Config Name="certs" Target="/certs" Default="" Mode="rw" Description="Place your certs here after grabbing them off your device. (Yes you need to do this)" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/certs</Config>
+    <Config Name="library" Target="/library" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/library</Config>
+    <Config Name="firmware" Target="/firmware" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/data/firmware</Config>
```
Comment (nwithan8) [inline templates/teddycloud.xml:22]:
```suggestion
    <Config Name="firmware" Target="/teddycloud/data/firmware" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/data/firmware</Config>
```

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: templates/teddycloud.xml (hunk 4 above)
Comment (nwithan8) [reply templates/teddycloud.xml:21]: Refer to: https://github.com/selfhosters/unRAID-CA-templates/pull/443#discussion_r1434741027

File: templates/teddycloud.xml (hunk 3 above)
Comment (nwithan8) [reply templates/teddycloud.xml:20]: Refer to: https://github.com/selfhosters/unRAID-CA-templates/pull/443#discussion_r1434741027

File: templates/teddycloud.xml (hunk 6 above)
Comment (nwithan8) [reply templates/teddycloud.xml:22]: Refer to: https://github.com/selfhosters/unRAID-CA-templates/pull/443#discussion_r1434741027

File: templates/teddycloud.xml (hunk 2 above)
Comment (nwithan8) [reply templates/teddycloud.xml:19]: Refer to: https://github.com/selfhosters/unRAID-CA-templates/pull/443#discussion_r1434741027

**Comment by catduckgnaf** [comment]
I believe I made all required and suggested changes.

**Review by nwithan8 — APPROVED** [review]
Looks good to me, good work!

File: templates/teddycloud.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    <Privileged>false</Privileged>
+    <Support>https://t.me/toniebox_reverse_engineering</Support>
+    <Project>https://github.com/toniebox-reverse-engineering/teddycloud</Project>
+    <Overview>teddyCloud is an alternative server for your Toniebox, allowing you to host the cloud services locally. This gives you the control about which data is sent to the original manufacturers cloud and allows you to host your own figurine audio files on e.g. your NAS or any other server.
+    </Overview>
+    <WebUI>http://[IP]:[PORT:7780]</WebUI>
+    <Category>MediaServer:Music MediaServer:Other</Category>
+    <TemplateURL>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/teddycloud.xml</TemplateURL>
+    <Icon>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/img/teddycloud.xml</Icon>
+    <Config Name="WEB UI" Target="80" Default="7780" Mode="tcp" Description="This an optional port for the webui of teddycloud" Type="Port" Display="always" Required="true" Mask="false">7780</Config>
```
Comment (Roxedus) [inline templates/teddycloud.xml:16]:
```suggestion
    </Config Name="WEB UI" Target="80" Default="7780" Mode="tcp" Description="This an optional port for the webui of teddycloud" Type="Port" Display="always" Required="true" Mask="false">
```
No need to set the value when using Default, applies to all Config's

**Review by Roxedus — CHANGES_REQUESTED** [review]

File: templates/teddycloud.xml (hunk 7 above)
Comment (catduckgnaf) [reply templates/teddycloud.xml]: got it, fixed.

File: templates/teddycloud.xml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    <Config Name="config" Target="/teddycloud/config" Default="" Mode="rw" Description="" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/teddycloud/config</Config>
+    <Config Name="certs" Target="/teddycloud/certs" Default="" Mode="rw" Description="Place your certs here after grabbing them off your device. (Yes you need to do this)" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/certs</Config>
+    <Config Name="library" Target="/teddycloud/data/library" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="false" Mask="false">/mnt/user/appdata/teddycloud/data/library</Config>
+    <Config Name="firmware" Target="/teddycloud/data/firmware" Default="" Mode="rw" Description="" Type="Path" Display="advanced" Required="true" Mask="false">/mnt/user/appdata/teddycloud/data/firmware</Config>
+    <Config Name="PUID" Target="PUID" Default="" Mode="" Description="" Type="Variable" Display="advanced" Required="false" Mask="false">99</Config>
+    <Config Name="PGID" Target="PGID" Default="" Mode="" Description="" Type="Variable" Display="advanced" Required="false" Mask="false">100</Config>
```
Comment (Roxedus) [inline templates/teddycloud.xml:24]: These dont seem to be supported, looking at the dockerfile

**Review by Roxedus — APPROVED** [review]

**Pull request merged**